test/kmeans.jl

```diff
@@ -32,6 +32,15 @@ end
     end
 end
 
+@testset "k=1 and k=n corner cases" begin
+    Random.seed!(34568)
+    @test length(kmeans(randn(2,3), 1).centers) == 2
+    @test length(kmeans(randn(2,3), 3).centers) == 6
+    x = [[ 0.5 1 ]; [ 1 0.5 ]]
```

Comment:
Simpler form:
```suggestion
    x = [0.5 1; 1 0.5]
```

What about 3-point example? I think we have more chances to detect when something is wrong with 3 points:
```suggestion
    x = [0.5 1 2; 1 0.5 0]
```

Reply:
Looking at it, the _old_ version also accepts k=2 in a 2x? matrix input; I'll use a 3x3 instead.

Reply:
Hmm, that's strange. The columns are the datapoints, so it should throw for *k=2* and 2x1 or 2x2 *x* matrix.